src/main/java/frc/robot/utils/ThreadedGyro.java

```diff
@@ -0,0 +1,56 @@
+package frc.robot.utils;
+
+import com.kauailabs.navx.frc.AHRS;
+import frc.robot.constants.Constants;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ThreadedGyro {
+    private final AHRS gyro;
+    private final AtomicBoolean shouldReset = new AtomicBoolean(false);
+    private final AtomicBoolean shouldOffset = new AtomicBoolean(false);
+    private final AtomicLong lastGyro;
+    private final AtomicLong gyroOffset = new AtomicLong();
+    private final ScheduledExecutorService executor;
+
+    public ThreadedGyro(AHRS gyro) {
+        this.gyro = gyro;
+        this.lastGyro = new AtomicLong((Double.doubleToLongBits(0)));
+        this.executor = Executors.newScheduledThreadPool(1);
+    }
+
+    public void start(){
+        executor.scheduleAtFixedRate(() -> {
+            if (shouldReset.get()) {
+                gyro.reset();
+                shouldReset.set(false);
+            }
+            if (shouldOffset.get()){
+                gyro.setAngleAdjustment(Double.longBitsToDouble(gyroOffset.get()));
+                shouldReset.set(false);
```

Comment:
`shouldOffset.set`

Reply:
kind of defeats the whole purpose if I don't set it to false idk how i missed that in new PR